general/README.md

```diff
@@ -0,0 +1,23 @@
+# General Guidelines
+
+The general guidelines provide an overview of sustainable software development and good scientific practice in research software engineering. They provide a general framework which researchers should adhere to when writing scientific software. However, each programming language introduces its own flavor and so the tools that are recommended can be language-specific. Please check the guidelines for the language that you are using. Currently, [C++](../cpp/README.md) and [python](../python/README.md) are available. Please open an issue for requesting recommendations for further programming languages.
+
+## Version control
+
+Any software that is being actively developed should be under version control. Version control allows you to keep track of any changes to your software and review the history at any time. It also allows you to track issues and work collaboratively. A version-control system for your software should be the default. Different systems are available and at this point we recommend [github](www.github.com).
+
+## Documentation
+
+A good documentation is detrimental for ensuring users know what to do with the code, and applying it to the correct use cases. It will save time in the long run, as code by itself is rarely obvious, and it will help outline dependencies. Depending on the programming language used, we recommend different tools like `doxygen`, `sphinx`, or using markdown/a wiki.
```

Comment:
```suggestion
Good documentation is essential for ensuring users know what to do with the code, and are applying it to the correct use cases. It will save time in the long run, as code by itself is rarely obvious, and it will help outline dependencies. Depending on the programming language used, we recommend different tools like `doxygen`, `sphinx`, or using markdown/a wiki.
```